frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx

```diff
@@ -1,208 +1,340 @@
 import InfoBanner from "components/InfoBanner/InfoBanner";
 import PoliciesListWrapper from "./components/PoliciesListWrapper";
 import AddPolicyModal from "./components/AddPolicyModal";
 import RemovePoliciesModal from "./components/RemovePoliciesModal";
+import TeamsDropdown from "./components/TeamsDropdown";
 
 const baseClass = "manage-policies-page";
 
 const DOCS_LINK =
-  "https://fleetdm.com/docs/deploying/configuration#osquery_policy_update_interval";
-interface IRootState {
-  app: {
-    config: IConfig;
-  };
-  entities: {
-    queries: {
-      isLoading: boolean;
-      data: IQuery[];
-    };
-  };
-}
-
-const renderTable = (
-  policiesList: IPolicy[],
-  isLoading: boolean,
-  isLoadingError: boolean,
-  onRemovePoliciesClick: (selectedTableIds: number[]) => void,
-  toggleAddPolicyModal: () => void
-): JSX.Element => {
-  if (isLoadingError) {
-    return <TableDataError />;
-  }
+  "https://fleetdm.com/docs/deploying/configuration#osquery_detail_update_interval";
 
-  return (
-    <PoliciesListWrapper
-      policiesList={policiesList}
-      isLoading={isLoading}
-      onRemovePoliciesClick={onRemovePoliciesClick}
-      toggleAddPolicyModal={toggleAddPolicyModal}
-    />
-  );
+const renderInheritedPoliciesButtonText = (
+  showPolicies: boolean,
+  policies: IPolicy[]
+) => {
+  const count = policies.length;
+
+  return `${showPolicies ? "Hide" : "Show"} ${count} inherited ${
+    count > 1 ? "policies" : "policy"
+  }`;
 };
 
-const ManagePolicyPage = (): JSX.Element => {
+const ManagePolicyPage = (managePoliciesPageProps: {
+  router: any;
+  location: any;
+}): JSX.Element => {
+  const { location, router } = managePoliciesPageProps;
+
   const dispatch = useDispatch();
 
-  const queries = useSelector((state: IRootState) => state.entities.queries);
-  const queriesList = Object.values(queries.data);
+  const {
+    config,
+    currentUser,
+    isAnyTeamMaintainer,
+    isGlobalAdmin,
+    isGlobalMaintainer,
+    isOnGlobalTeam,
+    isFreeTier,
+    isPremiumTier,
+  } = useContext(AppContext);
 
+  const { isTeamMaintainer } = permissionsUtils;
+  const canAddOrRemovePolicy = (user: IUser | null, teamId: number | null) =>
+    isGlobalAdmin || isGlobalMaintainer || isTeamMaintainer(user, teamId);
+
+  const { data: teams } = useQuery(["teams"], () => teamsAPI.loadAll({}), {
+    enabled: !!isPremiumTier,
+    select: (data) => data.teams,
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
+  });
+
+  const { data: fleetQueries } = useQuery(
+    ["fleetQueries"],
+    () => fleetQueriesAPI.loadAll(),
+    {
+      select: (data) => data.queries,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  // ===== local state
+  const [globalPolicies, setGlobalPolicies] = useState<IPolicy[] | never[]>([]);
+  const [isLoadingGlobalPolicies, setIsLoadingGlobalPolicies] = useState(true);
+  const [isGlobalPoliciesError, setIsGlobalPoliciesError] = useState(false);
```

Comment:
If we need to keep track of loading and errors, then this is a candidate for `useQuery` from the `react-query` library. And if we need to control when to refetch the data, try using the `refetch` prop provided.

Docs: https://react-query.tanstack.com/reference/useQuery